packages/itwinui-react/vitest.config.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  test: {
+    globals: true,
+    environment: 'jsdom',
+    setupFiles: 'setupTests.ts',
+    coverage: {
+      provider: 'v8',
+    },
+    exclude: ['/node_modules/', '/esm/', '/cjs/'],
+    reporters: ['junit', 'default'],
+    outputFile: 'coverage/junit.xml',
```

Comment:
Any reason you chose this specific set of coverage options? Can't we use default `coverage` options?

And have you verified that coverage is working as expected? There are two places we used to show coverage:
- Table displayed when running in the CLI
- A Github check displayed when running in CI

While you're at it, you can also bump the github action `EnricoMi/publish-unit-test-result-action` to the latest version.

Reply:
When I run `yarn workspace @itwin/itwinui-react test:unit`, I don't get the coverage report that we used to get before. Is `coverage: { enabled: true }` required?

<img width="465" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/f5ed882d-7507-4ffc-a71c-7467d2e6da4d">

Reply:
Maybe we can change the `test:unit` command from `vitest run` to `vitest run --coverage`.

Reply:
> Added coverage flag as ([36c16d7](https://github.com/iTwin/iTwinUI/commit/36c16d797e98f5f9ba8b5ebebf76ff9c9f2b2f91)) but in the [CI](https://github.com/iTwin/iTwinUI/actions/runs/7672013541/job/20911611358?pr=1784), it was getting added twice like this `vitest run --coverage --coverage`. So I added `coverage.enabled=true` and it works the same, so keeping this.

Why not remove `--coverage` from the github workflow then? Using the `--coverage` feels more correct than the `coverage.enabled` config option, because we are opting into it rather than always having it enabled.

Also still waiting for this:
 
> While you're at it, you can also bump the github action `EnricoMi/publish-unit-test-result-action` to the latest version.

Reply:
> Done ([04958b5](https://github.com/iTwin/iTwinUI/commit/04958b5d40b2521c3a5f9e85957b4089bd7b7e17))

I feel coverage should be opt-in in `package.json`'s `test:unit` script too. For e.g., if I am trying to pass just one failing unit test, the coverage report is unnecessary and obstructive.

Reply:
Yes